theano/scan_module/scan_op.py

```diff
@@ -2198,7 +2200,10 @@ def compute_all_gradients(known_grads):
                 dC_dinps_t[dx + self.n_seqs] += dC_dXtm1
         # Construct scan op
         # Seqs
-        outer_inp_seqs = [x[::-1] for x in inputs[1:1 + self.n_seqs]]
+        if self.as_while:
+            outer_inp_seqs = [x[n_steps - 1::-1] for x in inputs[1:1 + self.n_seqs]]
```

Comment:
Add a comment to what is the equivalent, more readable version of x[n_steps - 1::-1].